src/window/tumbling_window.rs

```diff
@@ -131,16 +133,17 @@ impl Windower for TumblingWindower {
         closed_ids
     }
 
-    fn activate_after(&mut self, watermark: &NaiveDateTime) -> Option<Duration> {
-        let watermark = *watermark;
-        self.close_times
-            .values()
-            .cloned()
-            .map(|t| t - watermark)
-            .min()
+    fn fate(&self) -> WindowerFate {
+        if let Some(next_close) = self.close_times.values().cloned().min() {
```

Comment:
why did i think these were sorted already? 🤔

Reply:
Maybe you're remembering that we use `BTreeMap` elsewhere? We can't here because we need to order by value, not key. We could cache the min close time, but I don't know that it would help that much.